cdap-ui/server/express.js

```diff
@@ -471,6 +449,44 @@ function makeApp(authAddress, cdapConfig, uiSettings) {
     },
   ]);
 
+  app.all('/api/:path(*)', function(req, res) {
+    var opts = {
+      method: req.method,
+      url: urlhelper.constructUrl(cdapConfig, req.param('path')),
+      headers: {},
```

Comment:
As I had noted in the comment below passing in all the headers from `req` to `request` seem to be causing the stream that we pipe to never end. The handpicking of headers is to avoid that for all request methods.

Reply:
But that means if the rest call actually take other headers, then this will not work